### PR TITLE
Set up the account creation

### PR DESCRIPTION
**Pull request by yostyle**
#6

### CODE REVIEW

File: vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterFragment.kt [hunk 1]
```diff
@@ -74,9 +76,16 @@ class TchapRegisterFragment @Inject constructor(private val platformViewModelFac
...
-        loginViewModel.observeViewEvents {
-            when (it) {
+        loginViewModel.observeViewEvents { loginViewEvents ->
+            when (loginViewEvents) {
                 TchapLoginViewEvents.OnLoginFlowRetrieved -> loginViewModel.handle(TchapLoginAction.LoginOrRegister(login, password, getString(R.string.login_default_session_public_name)))
+                is TchapLoginViewEvents.RegistrationFlowResult -> {
+                    val emailStage = loginViewEvents.flowResult.missingStages.firstOrNull { it.mandatory && it is Stage.Email }
+
+                    if (emailStage != null) {
+                        loginViewModel.handle(TchapLoginAction.AddThreePid(RegisterThreePid.Email(login)))
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterFragment.kt:86]: I would prefer to check whether login is actually an email address before using it
(Do you know whether element-android check email string before using it)

File: vector/src/main/java/fr/gouv/tchap/features/login/TchapLoginViewModel.kt [hunk 2]
```diff
@@ -250,9 +237,9 @@ class TchapLoginViewModel @AssistedInject constructor(
         reAuthHelper.data = action.password
         currentJob = executeRegistrationStep {
             it.createAccount(
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/login/TchapLoginViewModel.kt:239]: We may add a comment here to explain:
- Tchap is not expected username
- this first register request will linked the pwd with the returned session id (used in the following steps)

File: vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterWaitForEmailFragment.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 New Vector Ltd
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterWaitForEmailFragment.kt:2]:
```suggestion
 * Copyright 2021 New Vector Ltd
```

File: vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterFragment.kt [hunk 4]
```diff
@@ -78,17 +79,25 @@ class TchapRegisterFragment @Inject constructor(private val platformViewModelFac
...
                     if (emailStage != null) {
                         loginViewModel.handle(TchapLoginAction.AddThreePid(RegisterThreePid.Email(login)))
-                    } else Unit
+                    } else {
+                        AlertDialog.Builder(requireActivity())
+                                .setTitle(R.string.dialog_title_error)
+                                .setMessage(R.string.login_error_unable_register)
+                                .setPositiveButton(R.string.ok, null)
+                                .show()
+                        Unit
```
Comment (giomfo) [inline vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterFragment.kt:96]: Don't you think we can remove this Unit?

**Review by giomfo — APPROVED** [review]
LGTM
Just 2 comments

File: vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterFragment.kt (hunk 4 above)
Comment (yostyle) [reply vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterFragment.kt:96]: No it's normal

**Pull request merged**